lib/puppet/functions/snmp/snmpv3_usm_hash.rb

```diff
@@ -19,24 +20,16 @@
   # @return [String] The calculated hash.
   #
   dispatch :snmpv3_usm_hash do
-    required_param 'String', :authtype
+    required_param "Enum['SHA','MD5']", :authtype
     required_param 'String', :engine
-    required_param 'String', :passphrase
+    required_param 'String[8]', :passphrase
     optional_param 'Integer', :bits
     return_type 'String'
   end
 
   def snmpv3_usm_hash(authtype, engine, passphrase, bits = nil)
     require 'digest'
 
-    unless passphrase.length >= 8
```

Comment:
If the function is private, I don't think we should ever be hitting this code anyway as the defined type already checks the string length.